editor/src/components/inspector/inspector-common.tsx

```diff
@@ -87,17 +105,48 @@ export function filterKeepFlexContainers(
   )
 }
 
-export function detectFlexDirection(
+const DefaultFlexDirection: FlexDirection = 'row'
+
+function detectFlexDirectionOne(
   metadata: ElementInstanceMetadataMap,
   elementPath: ElementPath,
 ): FlexDirection {
   const element = MetadataUtils.findElementByElementPath(metadata, elementPath)
-  if (element == null) {
-    return 'row'
+  if (element == null || !MetadataUtils.isFlexLayoutedContainer(element)) {
+    return DefaultFlexDirection
   }
 
-  return stringToFlexDirection(element.computedStyle?.['flexDirection'] ?? null) ?? 'row'
+  return (
+    stringToFlexDirection(element.computedStyle?.['flexDirection'] ?? null) ?? DefaultFlexDirection
+  )
+}
+
+export function detectFlexDirection(
+  metadata: ElementInstanceMetadataMap,
+  elementPaths: Array<ElementPath>,
+): FlexDirection {
+  const allDetectedMeasurement = elementPaths.map((path) => detectFlexDirectionOne(metadata, path))
+  return allElemsEqual(allDetectedMeasurement, (l, r) => l === r)
+    ? allDetectedMeasurement[0]
+    : DefaultFlexDirection
+}
+
+export function detectAreElementsInFlexLayout(
+  metadata: ElementInstanceMetadataMap,
+  elementPaths: Array<ElementPath>,
+): boolean {
+  return elementPaths.every((path) =>
+    MetadataUtils.isFlexLayoutedContainer(MetadataUtils.findElementByElementPath(metadata, path)),
```

Comment:
this is actually wrong! there's `MetadataUtils.isParentFlexLayoutedContainerForElement()` that should be used if you are talking about whether the element at path is _in_ a flex parent

but if you want to detect if it's in a flex parent -and- participates in the layout, we have the incredibly named `MetadataUtils.isParentYogaLayoutedContainerForElementAndElementParticipatesInLayout`!

Reply:
(discussed this and it turned out that the name of the function was misleading, I renamed it to `detectAreElementsFlexContainers`)